### DISCUSSION TITLE
Integrate the "New State-of-the-Art TTS Model Released: F5-TTS"

### DISCUSSION THREAD

**Discussion by rwenckens** [aedocw/epub2tts#270] [Ideas]
Playing around with their demo from https://www.reddit.com/r/StableDiffusion/comments/1g2giso/new_stateoftheart_tts_model_released_f5tts/
I was amazed at the quality and the similarity to the reference audio.

**Comment by aedocw** [comment]
Wow that does sound *really* good.
I'll play with this some time over the next few weeks and see what it would take to incorporate. Thanks for sharing!

**Comment by danielw97** [comment]
Hi Aedocw,
I'm Still using this great utility regularly and stumbled across this discussion.
I've been following f5tts development closely over the last week or so, and they do now have a pip installable package.
I don't think it's been pushed to pyPi yet, but it can be seen on the main branch.
There's also an API script or cli inference that could potentially be used as well.
https://github.com/swivid/f5-TTS/
I'm happy to test things depending on if anyone is able to do any work on this as well, and will be keeping an eye here.

**Comment by danielw97** [comment]
Hi again,
just a quick note to say that alltalk tts (which has been mentioned in the past on this repo) now has an openai compatible endpoint in the beta version of v2, which is available on the main repo under the alltalkbeta branch.
There are 4 tts engines already including xtts, piper, parler and f5tts which is the reason I bring this up.
I've not had time to test how this works with epub2tts yet, although hope this may be a useable solution to decrease the workload of adding each individual tts engine to this utility.
When I have time I'll report back, maybe opening a separate discussion if there are any problems.

**Comment by danielw97** [comment]
From an initial look, the alltalk api seems to work out of the box due to the fact that epub2tts is splitting by sentances now.
If anyone's interested in trying this, you have to specify your OPENAI_API_BASE to your alltalk endpoint.
in my case it is http://127.0.0.1:7851/v1
Make sure to set threads to 1 when using this, as it can't process in parallel.
